host/src/provider_db.rs

```diff
@@ -224,16 +230,35 @@ impl Database for ProviderDb {
             return Ok(block_hash);
         }
 
-        // Get the 256 history block hashes from the provider at first time for anchor
-        // transaction.
         let block_number = u64::try_from(number).unwrap();
-        for block in batch_get_history_headers(&self.provider, &self.async_executor, block_number)?
-        {
-            let block_number = block.header.number.unwrap().try_into().unwrap();
-            let block_hash = block.header.hash.unwrap();
+        if self.network.is_taiko() {
```

Comment:
ah, last time I saw John's PR here I wonder if we need to support non-taiko provider, and I convinced myself that raiko is for taiko only so the 256 batch api is always there.
So now this assumption is not valid anymore? we make raiko a general type1 (if this is still meaningful) ethereum evm emulator to support not only taiko?

Reply:
I would say as far as reasonable and if only a few code changes required, I would try to make it general type 1. Mainly because there's so much testing data available for Ethereum and the changes to support it currently are so minimal. The current code also does not support it 100% correctly (some additional checks would be required), but good enough to not fail.

Also a good way to test future Ethereum versions like cancun before Taiko actually supports them!